files/conf/90-storaged.rules.in

```diff
@@ -17,7 +17,7 @@
 # LICENSE@@@
 
 
-KERNEL=="usb_gadget", SUBSYSTEM=="platform", DRIVER=="usb_gadget", ENV{G_SUBSYSTEM}=="storage", ACTION=="change", RUN+="@WEBOS_INSTALL_SYSCONFDIR@/udev/scripts/storage.sh $ENV{G_ACTION} $ENV{G_MEDIA_LOADED}$ENV{G_HOST_CONNECTED}$ENV{G_MEDIA_REQUESTED}$ENV{G_BUS_SUSPENDED}"
```

Comment:
@WEBOS_INSTALL_SYSCONFDIR@ should be replaced by during build by /etc. So the actual issue might be just the missing /bin/sh somehow? Really not sure here. Seems legacy had the following:

`# route mass storage events to storaged over lunaservice bus
KERNEL=="usb_gadget", SUBSYSTEM=="platform", DRIVER=="usb_gadget", ENV{G_SUBSYSTEM}=="storage", ACTION=="change", RUN+="/etc/udev/scripts/storage.sh $ENV{G_ACTION} $ENV{G_MEDIA_LOADED}$ENV{G_HOST_CONNECTED}$ENV{G_MEDIA_REQUESTED}$ENV{G_BUS_SUSPENDED}"`

Reply:
The syntax is just wrong.  Per the manual page, https://man7.org/linux/man-pages/man7/udev.7.html

Search for "The RUN substitutions are performed after all rules", to find the correct syntax for substitution inside a field.  The ENV{} syntax is correct for G_SUBSYSTEM but $env{key} is correct for inside the RUN.